File: examfront/src/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {


  constructor( private http : HttpClient) { }

  public getAllQuiz()
  {
    return  this.http.get(`${baseUrl}/quiz/all-quizzes`);
  }

  public addQuiz( quiz : any){
    return this.http.post(`${baseUrl}/quiz/add-quiz`, quiz);
  }
  public deleteQuiz(quizId : any)
  {
    console.log("deleting");
    return this.http.delete(`${baseUrl}/quiz/${quizId}`);
  }
  // get single quiz

  public getSingleQuiz(qId : any)
  {
   // console.log(`${baseUrl}/quiz/${qId}`);
    return this.http.get(`${baseUrl}/quiz/${qId}`);
  }
  public updateQuiz(quiz : any)
  {
    console.log("Quiz adding")
    return this.http.put(`${baseUrl}/quiz/update-quiz`,quiz);
  }
  
  // Get quizzess of category
  public getQuizzessOfCategory(cid : any)
  {
    return this.http.get(`${baseUrl}/quiz/category/${cid}`)
  } 

  // Get active Quizzess
  public getActiveQuizzess(){
    return this.http.get(`${baseUrl}/quiz/active`);
  }

  // Get quizzess by category and active
  public getQuizzessOfActiveCategory(cid : any){
    return this.http.get(`${baseUrl}/quiz/category/active/${cid}`);
  }
}

File: JwtExample3/JwtExample3/src/main/java/com/jwt/example/service/QuizService.java
package com.jwt.example.service;

import java.util.List;
import java.util.Set;

import com.jwt.example.entity.exam.Category;
import com.jwt.example.entity.exam.Quiz;

public interface QuizService {
	public Quiz addQuiz(Quiz q);
	public Quiz updateQuiz(Quiz q);
	public Set<Quiz> getQuizzes();
	public Quiz getQuiz(Long qId);
	public void deleteQuiz(Long qId);
	public List<Quiz>  getQuizzessofCategory(Category c);
	
	// Get Active quizzess
	public List<Quiz> getActiveQuiz();
	
	// Get quiz by category and and active
	public List<Quiz> getQuizByCategoryAndActive(Category c );

}
